### ISSUE TITLE
obspy couldn't find a station in IRIS DMC

### ISSUE THREAD

**Issue by CovMat** [obspy/obspy#2031]
Hi
I use the latest obspy under python 3.6.0 and CentOS 6.9 x64:
```
$ python --version
Python 3.6.0 :: Anaconda custom (64-bit)
$ lsb_release -a
LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch
Distributor ID:	CentOS
Description:	CentOS release 6.9 (Final)
Release:	6.9
Codename:	Final
```

I try to download waveform of DK.DY2G using following python script:
```
import obspy
from obspy.clients.fdsn.mass_downloader import RectangularDomain, \
    Restrictions, MassDownloader
import os

domain = RectangularDomain(minlatitude=60, maxlatitude=80,
                           minlongitude=-75, maxlongitude=-15)

restrictions = Restrictions(
    
    starttime=obspy.UTCDateTime(2012, 1, 1, 0, 0, 0),
    endtime=obspy.UTCDateTime(2012, 2, 1, 0, 0, 0),
    
    network="DK", channel="BHZ", station="DY2G", location="01",
    reject_channels_with_gaps=False,
    sanitize=True,
    minimum_length=0.0,
    minimum_interstation_distance_in_m=100.0)

mdl = MassDownloader(providers=["IRIS"])
def get_mseed_storage(network, station, location, channel, starttime, endtime):
    return os.path.join("waveforms/", "%s.%s.%s.%s.mseed" % (network, station, location, channel))
mseed_storage = get_mseed_storage
tt = mdl.download(domain, restrictions, mseed_storage, stationxml_storage="stations")
```
However, obspy couldn't find this station:
```
[2017-12-24 20:16:01,445] - obspy.clients.fdsn.mass_downloader - INFO: Initializing FDSN client(s) for IRIS.
[2017-12-24 20:16:07,360] - obspy.clients.fdsn.mass_downloader - INFO: Successfully initialized 1 client(s): IRIS.
[2017-12-24 20:16:07,360] - obspy.clients.fdsn.mass_downloader - INFO: Total acquired or preexisting stations: 0
[2017-12-24 20:16:07,360] - obspy.clients.fdsn.mass_downloader - INFO: Client 'IRIS' - Requesting reliable availability.
[2017-12-24 20:16:07,975] - obspy.clients.fdsn.mass_downloader - INFO: Client 'IRIS' - Successfully requested availability (0.61 seconds)
[2017-12-24 20:16:07,976] - obspy.clients.fdsn.mass_downloader - INFO: Client 'IRIS' - Found 0 stations (0 channels).
[2017-12-24 20:16:07,977] - obspy.clients.fdsn.mass_downloader - INFO: Client 'IRIS' - No data available.
[2017-12-24 20:16:07,977] - obspy.clients.fdsn.mass_downloader - INFO: ============================== Final report
[2017-12-24 20:16:07,977] - obspy.clients.fdsn.mass_downloader - INFO: 0 MiniSEED files [0.0 MB] already existed.
[2017-12-24 20:16:07,977] - obspy.clients.fdsn.mass_downloader - INFO: 0 StationXML files [0.0 MB] already existed.
[2017-12-24 20:16:07,977] - obspy.clients.fdsn.mass_downloader - INFO: Client 'IRIS' - Acquired 0 MiniSEED files [0.0 MB].
[2017-12-24 20:16:07,977] - obspy.clients.fdsn.mass_downloader - INFO: Client 'IRIS' - Acquired 0 StationXML files [0.0 MB].
[2017-12-24 20:16:07,977] - obspy.clients.fdsn.mass_downloader - INFO: Downloaded 0.0 MB in tota
```
But the data of this station exist and are available in IRIS DMC:
[http://ds.iris.edu/mda/DK/DY2G/01/BHZ?timewindow=2011/05/19%2000:00:00-2012/06/07%2022:59:59](http://ds.iris.edu/mda/DK/DY2G/01/BHZ?timewindow=2011/05/19%2000:00:00-2012/06/07%2022:59:59)

I think it may be a bug of obspy. Could you please check it and fix if possible. Thanks.

**Comment by ThomasLecocq** [comment]
Indeed, this looks like a bug. It seems the default ``location_priorities`` are still used although the ``Restrictions().location`` is defined, which results in an empty ``filtered_channels`` list at this line 

https://github.com/obspy/obspy/blob/master/obspy/clients/fdsn/mass_downloader/download_helpers.py#L1190

a workaround for you now is to add the:

```python
location_priorities=["01",],
```
to your ``Restrictions`` object.

**Comment by trichter** [comment]
Thanks for reporting!

Adding to Thomas: Another incarnation of this bug is described in #1810. A fix is prepared in #2028.

**Comment by bobonice** [comment]
👍 Thanks !   Yesterday I found my mass_downloader script cannot download the waveform data of "DPZ" channel(such as 1B networks). 
With the similar solution, I add the code :
`channel_priorities=("DPZ",)` in my script ! It works now!

**Comment by megies** [comment]
Was this fixed by #2028 and/or #2047?

**Comment by megies** [comment]
fixed by linked PRs
